### PR TITLE
Add commands to create and delete remediations

### PR DESCRIPTION
**Pull request by RaphaelBut**
### What type of PR is this?

feature

### What this PR does / Why we need it?
Adds commands to create and delete RBAC for CAD investigations.
This will enable CAD to access kube-api with per remediation scoped down permissions. 
These commands are only allowed for backplane roles which are configured to use remediations ( currently CAD and SREP, see https://gitlab.cee.redhat.com/service/backplane-api/-/merge_requests/378/diffs#c6d8d5258b671d72ca173652ee8826e678b71b62_10_14 )

To learn more about the remediation endpoint in backplane-api, see  https://gitlab.cee.redhat.com/service/backplane-api/-/merge_requests/378
### Which Jira/Github issue(s) does this PR fix?

https://issues.redhat.com/browse/OSD-25689

#### Guidelines
- If it's a new sub-command or new function to an existing sub-command, please cover at least 50% of the code
- If it's a bug fix for an existing sub-command, please cover 70% of the code 
 
#### Test coverage checks  
- [X] Added unit tests
- [ ] Created jira card to add unit test
- [ ] This PR may not need unit tests

### Pre-checks (if applicable)
- [X] Ran unit tests locally
- [X] Validated the changes in a cluster
- [ ] Included documentation changes with PR

### CODE REVIEW

File: pkg/utils/jwt.go [hunk 1]
```diff
@@ -36,3 +34,29 @@ func GetStringFieldFromJWT(token string, field string) (string, error) {
 
 	return claimString, nil
 }
+
+// GetUsernameFromJWT returns the username extracted from JWT token
+func GetUsernameFromJWT(token string) string {
```
Comment (samanthajayasinghe) [inline pkg/utils/jwt.go:39]: It's good , if we could write unit test to cover this func

File: pkg/utils/jwt.go [hunk 2]
```diff
@@ -36,3 +34,29 @@ func GetStringFieldFromJWT(token string, field string) (string, error) {
...
+	if !ok {
+		return "anonymous"
+	}
+	return claim.(string)
+}
+
+// GetContextNickname returns a nickname of a context
+func GetContextNickname(namespace, clusterNick, userNick string) string {
+	tokens := strings.SplitN(userNick, "/", 2)
+	return namespace + "/" + clusterNick + "/" + tokens[0]
```
[the remark this answers is not in the document]
Comment (RaphaelBut) [reply pkg/utils/jwt.go:61]: not exactly sure, it will most likely break your local kubeconfig context for a cluster :D 
this is from existing login.go and i would avoid changing its behavior in this PR if at all possible
I added test cases anyway, but they mostly highlight the problem

File: pkg/utils/jwt.go (hunk 1 above)
Comment (RaphaelBut) [reply pkg/utils/jwt.go:39]: sure thing ;) also moved from login.go without a change. Added test cases.

File: cmd/ocm-backplane/remediation/remediation.go [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+			return "", err
+		}
+		return "", errors.New(*dest.Message)
+
+	}
+	return api + *resp.JSON200.ProxyUri, nil
+}
+
+// CreateRemediationWithConn can be used to programtically interact with backplaneapi
+func CreateRemediationWithConn(bp config.BackplaneConfiguration, ocmConnection *ocmsdk.Connection, clusterID string, remediationName string) (config *rest.Config, serviceAccountName string, err error) {
```
[the remark this answers is not in the document]
Comment (RaphaelBut) [reply cmd/ocm-backplane/remediation/remediation.go]: Agreed! Moved to pkg/remediation
I did not move the unit test as they are mocking the command mostly and should reasonably cover the remediation pkg 

The Create|DeleteRemediationWithConn  functions are not tested currently but I think we do not need them actually.
While trying to test/mock/deal with them, I realize this is code leftover from an earlier draft for the same remediation feature. 
There is really no difference other then give the access token per parameter or give the ocm connection and then building a rest config instead of return the proxyUrl 
I think  this will need careful examination of what to export and what to use in CAD/osdctl which both use this as a dependecy ( which is already strange enough, there should be some kind of library and backplane-cli should pull from there as well ) https://github.com/openshift/osdctl/blob/f475456b95544fce30378f4a7684903e0626efb1/cmd/common/helpers.go#L7 

getting late, excuse the confusing comment, this is mostly to capture my thought for future 
TLDR will take another look :D

**Comment by samanthajayasinghe** [comment]
/tide/merge-method-squash
